### ISSUE TITLE
At-rest encryption of content using sqlcipher

### ISSUE THREAD

**Issue by lupine** [deltachat/deltachat-core#194]
deltachat-core stores data on disk, in an sqlite3 database. It also stores some blobs in an associated directory.

Within the sqlite3 database is sensitive data. This includes at least encryption keys, but may also include decrypted IMs (I'm uncertain about that).

It would be good to integrate sqlcipher: https://www.zetetic.net/sqlcipher/open-source/ as an alternative to the plaintext sqlite3 database. This encrypts the database at rest. To decrypt it, you have to provide a passphrase of some kind - this can be provided via idiomatic UI for each client implementation, and perhaps even saved to secure keyrings as an option.

I don't think this would cover the external blobs, which may also be sensitive and plaintext (I haven't checked). Assuming they are, it could be addressed in a follow-up by encrypting them with the same algorithm + passphrase as used with sqlcipher; or we could use the public/private keypair already in the database to encrypt the blobs with.

**Comment by hpk42** [comment]
just to also note here: app state is separated in android from other apps, and delta marks its state as "do not backup to cloud".  So there is no easy way to access delta's state in a non-rooted phone.  That being said it might still make sense to use sqlcipher or something similar if the risk is that a phone is taken, rooted and examined.

**Comment by lupine** [comment]
I agree, the current state is pretty good - especially when you take full-disk encryption into account. However, a few compromise scenarios are improved by this change (especially taking desktops and laptops into account ;) ), and it's quite low-effort, so I thought I'd document it. Priority is very low for me.

**Comment by r10s** [comment]
i am wondering if we should move all blob-files to a blob-table before targeting this issue. however i am not sure if this will have a bad impact to the sqlite-database that may be several gb large then.

**Comment by r10s** [comment]
> i am wondering if we should move all blob-files to a blob-table before targeting this issue. however i am not sure if this will have a bad impact to the sqlite-database that may be several gb large then.
> 
> if someone has experience with that, hints would be very welcome.

some voices to this are at #484
